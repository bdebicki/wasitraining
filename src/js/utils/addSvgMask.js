import SVG_MASK from './elementHandlers/svg';

export const svgType = 'http://www.w3.org/2000/svg';

export function addSVGmask({
	maskBgClass,
	maskId,
	maskShape,
	svgClass,
}) {
	const svg = document.createElementNS(svgType, 'svg');
	const defs = document.createElementNS(svgType, 'defs');
	const mask = document.createElementNS(svgType, 'mask');
	const rectMask = document.createElementNS(svgType, 'rect');
	const rectBg = document.createElementNS(svgType, 'rect');

	svg.classList.add(SVG_MASK.MASK_CLASS);
	if (svgClass) {
		svg.classList.add(svgClass);
	}
	mask.id = maskId;
	mask.setAttributeNS(null, 'maskUnits', 'userSpaceOnUse');
	rectMask.classList.add(SVG_MASK.MASK_BASE_CLASS);
	rectBg.classList.add(SVG_MASK.MASK_BG_CLASS);
	if (maskBgClass) {
		rectBg.classList.add(maskBgClass);
	}
	rectBg.setAttributeNS(null, 'mask', `url(#${maskId})`);

	mask.appendChild(rectMask);
	mask.appendChild(maskShape);
	defs.appendChild(mask);
	svg.appendChild(defs);
	svg.appendChild(rectBg);

	return svg;
}

