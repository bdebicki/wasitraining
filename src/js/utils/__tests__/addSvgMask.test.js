import { addSVGmask, svgType } from '../addSvgMask';
import using from '../../../../tests/utils/using';

function getMaskShape() {
	const shape = document.createElementNS(svgType, 'text');
	shape.textContent = '2018';

	return shape;
}

const svgMaskCases = [
	{
		description: 'render svg mask with additional classes dor svg and mackBg',
		settings: {
			svgClass: 'mainTimelineItemMask',
			maskId: 'yearMask2018',
			maskShape: getMaskShape(),
			maskBgClass: 'mainTimelineItemMask__bg',
		},
	},
	{
		description: 'render svg mask without addidional classes',
		settings: {
			maskId: 'yearMask2018',
			maskShape: getMaskShape(),
		},
	},
];

describe('render svg element', () => {
	using(svgMaskCases).describe('', ({ description, settings }) => {
		it(description, () => {
			const svgEl = addSVGmask(settings);

			expect(svgEl).toMatchSnapshot();
		});
	});
});
