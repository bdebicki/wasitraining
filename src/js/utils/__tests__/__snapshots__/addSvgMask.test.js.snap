// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render svg element	render svg mask with additional classes dor svg and mackBg 1`] = `
<svg
	class="svgMask mainTimelineItemMask"
>
	<defs>
		<mask
			id="yearMask2018"
			maskUnits="userSpaceOnUse"
		>
			<rect
				class="svgMask__base"
			/>
			<text>
				2018
			</text>
		</mask>
	</defs>
	<rect
		class="svgMask__bg mainTimelineItemMask__bg"
		mask="url(#yearMask2018)"
	/>
</svg>
`;

exports[`render svg element	render svg mask without addidional classes 1`] = `
<svg
	class="svgMask"
>
	<defs>
		<mask
			id="yearMask2018"
			maskUnits="userSpaceOnUse"
		>
			<rect
				class="svgMask__base"
			/>
			<text>
				2018
			</text>
		</mask>
	</defs>
	<rect
		class="svgMask__bg"
		mask="url(#yearMask2018)"
	/>
</svg>
`;
